blockchain/utxoviewpoint.go

```diff
@@ -515,72 +612,93 @@ func (view *UtxoViewpoint) disconnectTransactions(block *dcrutil.Block, stxos []
 				continue
 			}
 
-			// Ensure the spent txout index is decremented to stay in sync with
-			// the transaction input.
+			// Ensure the spent txout index is decremented to stay in sync with the
+			// transaction input.
 			stxo := &stxos[stxoIdx]
 			stxoIdx--
 
-			// When there is not already an entry for the referenced transaction
-			// in the view, it means it was fully spent, so create a new utxo
-			// entry in order to resurrect it.
+			// The legacy v2 spend journal format only stored the flags when the
+			// output was the last unspent output of the transaction.  As a result,
+			// when the information is missing, search for it by scanning all possible
+			// outputs of the transaction since it must be in one of them.
+			//
+			// It should be noted that this is quite inefficient, but it realistically
+			// will almost never run since all new entries include the information for
+			// all outputs and thus the only way this will be hit is if a long enough
+			// reorg happens such that a block with the old spend data is being
+			// disconnected.  The probability of that in practice is extremely low to
+			// begin with and becomes vanishingly small the more new blocks are
+			// connected.  In the case of a fresh database that has only ever run with
+			// the new v2 format, this code path will never run.
 			txIn := msgTx.TxIn[txInIdx]
-			originHash := &txIn.PreviousOutPoint.Hash
-			originIndex := txIn.PreviousOutPoint.Index
-			entry := view.entries[*originHash]
-			if entry == nil {
-				if !stxo.txFullySpent {
-					return AssertError(fmt.Sprintf("tried to revive unspent "+
-						"tx %v from non-fully spent stx entry", originHash))
+			if !stxo.IsSpent() {
```

Comment:
I know you are already familiar with this but just adding some context here.  The only case where `stxo.IsSpent()` would return `false` is for legacy v2 spend journal entries that were **not** the last spent output of the containing transaction (since the spent flag used to represent _fully_ spent).  This case is the only scenario where the `stxo` does not have the encoded flags stored, and therefore the flags need to be retrieved from a utxo for that transaction.